internal/config/config.go

```diff
@@ -8,98 +8,201 @@ import (
 type ControlPlane struct {
-	Enabled           bool                  `mapstructure:"enabled"`
-	ETCD              ControlPlaneComponent `mapstructure:"etcd"`
-	APIServer         ControlPlaneComponent `mapstructure:"apiServer"`
+	// Enabled controls whether control plane scraping will be attempted, for any component.
+	Enabled bool `mapstructure:"enabled"`
+	// ETCD contains configuration for the etcd scraper.
+	ETCD ControlPlaneComponent `mapstructure:"etcd"`
+	// APIServer contains configuration for the API server scraper.
+	APIServer ControlPlaneComponent `mapstructure:"apiServer"`
+	// ControllerManager contains configuration for the controller manager scraper.
 	ControllerManager ControlPlaneComponent `mapstructure:"controllerManager"`
-	Scheduler         ControlPlaneComponent `mapstructure:"scheduler"`
-	Timeout           time.Duration         `mapstructure:"timeout"`
-	Retries           int                   `mapstructure:"retries"`
+	// Scheduler contains configuration for the scheduler scraper.
+	Scheduler ControlPlaneComponent `mapstructure:"scheduler"`
+	// Timeout controls the timeout for the requests to control plane endpoints.
+	Timeout time.Duration `mapstructure:"timeout"`
+	// Retries controls how many times the integration will attempt to connect to control plane components before giving up.
+	Retries int `mapstructure:"retries"`
 }
 
+// ControlPlaneComponent contains the config for a control plane component.
 type ControlPlaneComponent struct {
-	Enabled        bool                       `mapstructure:"enabled"`
-	StaticEndpoint *Endpoint                  `mapstructure:"staticEndpoint"`
-	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
+	// Enabled controls whether this particular component should be scraped.
+	Enabled bool `mapstructure:"enabled"`
+	// StaticEndpoint contains an Endpoint configuration. If set, Autodiscover will not be attempted and the integration
+	// will contact this endpoint directly instead.
+	// Please note that failure to connect to a StaticEndpoint is considered a fatal error and will cause the
+	// integration to exit with a non-zero code.
+	StaticEndpoint *Endpoint `mapstructure:"staticEndpoint"`
+	// Autodiscover contains one or more criteria for discovering control plane endpoints. Entries will be iterated in
+	// order, with the following rules:
+	// 1. If an entry's criteria (Selector, Namespace, MatchNode) does not match any pod, the next entry will be tried.
+	// 2. If none of the entries matches any pod, the integration will not error but keep probing in case matching pods appear.
+	// 3. If an entry's criteria more than one pod, only the first match will be considered.
+	// matches an endpoint will be attempted.
+	// 4. Endpoints are tried in order for a matching pod, until metrics can be read successfully from one of them.
+	// 5. If all endpoints for a matching fail, no more entries will be processed, and the integration will error.
```

Comment:
the integration will not error in this case, what would happen is the same as point 2, the integration will start a new autodiscover cicle on the next iteration.